### PR TITLE
[5.1] Allow dynamic appending of attributes to query

### PR DESCRIPTION
**Pull request by adrianthedev**
We then could use it like this:

```
return $resource->findOrFail($id)
                      ->append(['custom_attribute'])
                      ->toArray();
```

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -735,6 +735,22 @@ public static function with($relations)
 
         return $instance->newQuery()->with($relations);
     }
+    
+    /**
+     * Append attributes to query when building a query
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: missing fullstop

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -735,6 +735,22 @@ public static function with($relations)
...
+     *
+     * @return $this
+     */
+    public function append($relations)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: missing phpdoc for this function

File: src/Illuminate/Database/Eloquent/Model.php [hunk 3]
```diff
@@ -735,6 +735,23 @@ public static function with($relations)
...
         return $instance->newQuery()->with($relations);
     }
+    
+    /**
+     * Append attributes to query when building a query.
+     * 
+     * @param  array|string  $relations
+     * @return $this
+     */
+    public function append($relations)
```
Comment (lukasgeiter) [inline src/Illuminate/Database/Eloquent/Model.php]: Calling the parameter `$relations` seems wrong to me though. Those could be dynamic attributes (via accessor) as well... How about `$attributes`?

Comment (adrianthedev) [inline src/Illuminate/Database/Eloquent/Model.php]: You're right! It's better this way.

**Comment by lukasgeiter** [comment]
:+1:

**Comment by adrianthedev** [comment]
Thank you for the help @GrahamCampbell and @lukasgeiter.
Thank you for the merge @taylorotwell.

**Pull request merged**